### ISSUE TITLE
AppendCharacterWithChar should consider constant type

### ISSUE THREAD

**Issue by Belle-PL** [pmd/pmd#3844]
**Affects PMD Version:6.42.0** 

**Rule:AppendCharacterWithChar**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_performance.html#appendcharacterwithchar>

**Description:**

**Code Sample demonstrating the issue:**
Hi, I found a false negative about the rule AppendCharacterWithChar. For the below code example, PMD should have reported a warning in the line 3. But it doesn't. Hence, I think this is a false negative. Thanks for your consideration.

```java
public void bar(StringBuffer sb) {
    final String a = "a";
    sb.append(a);
}
```

**Expected outcome:A warning**

PMD should report a violation at line 3, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*
